File: src/main/java/io/github/dylmeadows/eontimer/model/timer/Gen3TimerMode.java
package io.github.dylmeadows.eontimer.model.timer;

import io.github.dylmeadows.common.javafx.util.Choice;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum Gen3TimerMode implements Choice {
    STANDARD("Standard"),
    VARIABLE_TARGET("Variable Target");

    private final String text;
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/timer/Gen3TimerMode.kt
package io.github.dylmeadows.eontimer.model.timer

import io.github.dylmeadows.common.javafx.util.Choice

enum class Gen3TimerMode(private val text: String) : Choice {
    STANDARD("Standard"),
    VARIABLE_TARGET("Variable Target");

    override fun getText(): String {
        return text
    }
}